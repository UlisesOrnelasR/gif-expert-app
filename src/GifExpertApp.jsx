import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Puch", "Dagon ball"]);
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) {
      return;
    }
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory
        onNewCategory={(event) => {
          onAddCategory(event);
        }}
      />
      {categories.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </>
  );
};
