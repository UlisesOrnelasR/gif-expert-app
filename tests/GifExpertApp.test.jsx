import { render } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Prebas en el componente <GifExpertApp />", () => {
  test("Deberia hacer match con el snapshot", () => {
    const { contaienr } = render(<GifExpertApp />);
    expect(contaienr).toMatchSnapshot();
  });
});
