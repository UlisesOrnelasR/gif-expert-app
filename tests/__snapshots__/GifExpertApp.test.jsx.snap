// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prebas en el componente <GifExpertApp /> Deberia hacer match con el snapshot 1`] = `undefined`;
